--- a/arch/arm/boot/dts/dove.dtsi
+++ b/arch/arm/boot/dts/dove.dtsi
@@ -790,14 +790,14 @@
 			};
 
 			lcd1: lcd-controller@810000 {
-				compatible = "marvell,dove-lcd";
+				compatible = "marvell,dove-lcd", "marvell,armada-lcdc";
 				reg = <0x810000 0x1000>;
 				interrupts = <46>;
 				status = "disabled";
 			};
 
 			lcd0: lcd-controller@820000 {
-				compatible = "marvell,dove-lcd";
+				compatible = "marvell,dove-lcd", "marvell,armada-lcdc";
 				reg = <0x820000 0x1000>;
 				interrupts = <47>;
 				status = "disabled";
